app/controllers/LongerCommodityCodeController.scala

```diff
@@ -52,31 +52,48 @@ class LongerCommodityCodeController @Inject() (
 
   private val form = formProvider()
 
-  def onPageLoad(mode: Mode, recordId: String): Action[AnyContent] = (identify andThen getData andThen requireData) {
-    implicit request =>
-      val preparedForm = request.userAnswers.get(LongerCommodityCodePage(recordId)) match {
-        case None        => form
-        case Some(value) => form.fill(value)
-      }
+  def onPageLoad(mode: Mode, recordId: String): Action[AnyContent] = (identify andThen getData andThen requireData) { implicit request =>
+    val categorisationInfoOpt = request.userAnswers.get(RecordCategorisationsQuery).flatMap(_.records.get(recordId))
 
-      val commodityCodeOption = request.userAnswers
-        .get(RecordCategorisationsQuery)
-        .flatMap(x => x.records.get(recordId).map(x => x.commodityCode))
+    val originalComcodeOpt = categorisationInfoOpt.map(_.originalCommodityCode).getOrElse(None)
+    val latestComcodeOpt = categorisationInfoOpt.map(_.commodityCode)
 
-      commodityCodeOption match {
-        case Some(shortCommodity) if commodityCodeSansTrailingZeros(shortCommodity).length == 6 =>
-          Ok(view(preparedForm, mode, commodityCodeSansTrailingZeros(shortCommodity), recordId))
-        case _                                                                                  => Redirect(routes.JourneyRecoveryController.onPageLoad().url)
-      }
+    val latestDoesNotMatchOriginal = (latestComcodeOpt, originalComcodeOpt) match {
+      case (Some(latest), Some(original)) if latest != original.padTo(10, "0").mkString => true
+      case _ => false
+    }
+
+    val answerToFillFormWith = originalComcodeOpt.map { originalCode =>
+      latestComcodeOpt.getOrElse("").drop(originalCode.length)
+    }
+
+    val preparedForm = if (latestDoesNotMatchOriginal) {
+      answerToFillFormWith.map(form.fill).getOrElse(form)
+    } else {
+      form
+    }
+
+    originalComcodeOpt match {
+      case Some(shortCommodity) if commodityCodeSansTrailingZeros(shortCommodity).length == 6 =>
+        Ok(view(preparedForm, mode, commodityCodeSansTrailingZeros(shortCommodity), recordId))
+      case _ =>
+        Redirect(routes.JourneyRecoveryController.onPageLoad().url)
+    }
   }
 
   def onSubmit(mode: Mode, recordId: String): Action[AnyContent]           =
     (identify andThen getData andThen requireData).async { implicit request =>
-      val commodityCodeOption = request.userAnswers
-        .get(RecordCategorisationsQuery)
-        .flatMap(x => x.records.get(recordId).map(x => x.commodityCode))
+      val categorisationInfoOpt = request.userAnswers.get(RecordCategorisationsQuery).flatMap(_.records.get(recordId))
+
+      val originalComcodeOpt = categorisationInfoOpt.map(_.originalCommodityCode).getOrElse(None)
+      val latestComcodeOpt = categorisationInfoOpt.map(_.commodityCode)
 
-      commodityCodeOption match {
+      val latestDoesNotMatchOriginal = (latestComcodeOpt, originalComcodeOpt) match {
```

Comment:
some repeated code, maybe we can put these as functions in the Categorisation info object

Reply:
Moved to there :) committing soon